cellprofiler/modules/watershed.py

```diff
@@ -0,0 +1,109 @@
+"""
+
+Watershed
+
+"""
+
+import cellprofiler.image
+import cellprofiler.module
+import cellprofiler.object
+import cellprofiler.setting
+import skimage.measure
+import skimage.morphology
+
+
+class Watershed(cellprofiler.module.Module):
+    category = "Image segmentation"
+    module_name = "Watershed"
+    variable_revision_number = 1
+
+    def create_settings(self):
+        self.image_name = cellprofiler.setting.ImageNameSubscriber(
+            "Image"
+        )
+
+        self.object_name = cellprofiler.setting.ObjectNameProvider(
+            "Object",
+            ""
+        )
+
+        self.markers_name = cellprofiler.setting.ImageNameSubscriber(
+            "Markers"
+        )
+
+        self.mask_name = cellprofiler.setting.ImageNameSubscriber(
+            "Mask",
+            can_be_blank=True
+        )
+
+    def settings(self):
+        return [
+            self.image_name,
+            self.object_name,
+            self.markers_name,
+            self.mask_name
+        ]
+
+    def visible_settings(self):
+        return [
+            self.image_name,
+            self.object_name,
+            self.markers_name,
+            self.mask_name
+        ]
+
+    def run(self, workspace):
+        image_name = self.image_name.value
+
+        object_name = self.object_name.value
+
+        images = workspace.image_set
+
+        image = images.get_image(image_name)
+
+        dimensions = image.dimensions
+
+        image_data = image.pixel_data
+
+        markers_name = self.markers_name.value
+
+        markers = images.get_image(markers_name)
+
+        markers_data = markers.pixel_data
+
+        if not self.mask_name.is_blank:
+            mask_name = self.mask_name.value
+
+            mask = images.get_image(mask_name)
+
+            mask_data = mask.pixel_data
+        else:
+            mask_data = None
+
+        label_data = skimage.morphology.watershed(
+            image=image_data,
+            markers=markers_data,
+            mask=mask_data
+        )
+
+        label_data = skimage.measure.label(label_data)
```

Comment:
`skimage.measure.label` returns two values: `labels` and `num`. See http://scikit-image.org/docs/dev/api/skimage.measure.html#skimage.measure.label. You want the first return value.